doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,503 @@
+This proposal will follow the programming model of MPLS segment routing and IP tunnel termination by adding a LocalSID object for SRv6 similar to InSegment
+for MPLS.
+
+![SRv6 Endpoint Behavioral Model](figures/SRv6_Endpoint_behavioral_model.png "Figure 1: Endpoint Behavior ")
+__Figure 1: Endpoint Behavior.__
+
+### Source/Headend Behavior ###
+- This proposal will remove "Endpoint" from the Nexthop table attributes and leave other attributes like SIDLIST in order to implement the Headend Behavior.
+
+![SRv6 Headend Behavioral Model](figures/SRv6_Headend_behavioral_model.png "Figure 1: Headend Behavior ")
+__Figure 1: Headend Behavior.__
+
+## SAI Header changes ##
+
+### sainexthop.h
+
+1. sai_next_hop_endpoint_type_t and sai_nexthop_endpoint_pop_type_t is deprecated.
+
+```
+typedef enum _sai_next_hop_type_t
+{
+    ....
+    ....
+
+    /** IPv6 Segment Route SID List */
+    SAI_NEXT_HOP_TYPE_SEGMENTROUTE_SIDLIST,
+
+    /** IPv6 Segment Route Endpoint Function. **Deprecated and replaced by SAI_LOCAL_SID_ENTRY_ATTR object ***/
+    SAI_NEXT_HOP_TYPE_SEGMENTROUTE_ENDPOINT,
+    ...
+
+} sai_next_hop_type_t;
+
+
+**
+ * @brief Enum defining Endpoint Segment Pop types for End, End.X and End.T. **Deprecated
+ *        and replaced by sai_local_sid_entry_endpoint_flavor_t**
+ */
+
+typedef enum _sai_next_hop_endpoint_pop_type_t
+{
+    /** Penultimate segment pop */
+    SAI_NEXT_HOP_ENDPOINT_POP_TYPE_PSP,
+
+    /** Ultimate Segment pop */
+    SAI_NEXT_HOP_ENDPOINT_POP_TYPE_USP,
+
+} sai_next_hop_endpoint_pop_type_t;
+
+
+
+/**
+ * @brief Attribute id for next hop
+ */
+typedef enum _sai_next_hop_attr_t
+{
+...
+...
+
+    /**
+     * @brief Next hop entry Segment Route SID List
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_SEGMENTROUTE_SIDLIST
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_SEGMENTROUTE_SIDLIST
+     */
+    SAI_NEXT_HOP_ATTR_SEGMENTROUTE_SIDLIST_ID,
+
+    /**
+     * @brief Next hop entry Segment Route Endpoint Function. **Deprecated and replaced
+     * with SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE.**
+     *
+     * @type sai_next_hop_endpoint_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_SEGMENTROUTE_ENDPOINT
+    ** * @deprecated true**
+     */
+    SAI_NEXT_HOP_ATTR_SEGMENTROUTE_ENDPOINT_TYPE,
+
+    /**
+     * @brief Next hop entry Segment Route Endpoint Pop Option. **Deprecated and replaced
+     * with SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_FLAVOR**
+     *
+     * @type sai_next_hop_endpoint_pop_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @condition SAI_NEXT_HOP_ATTR_TYPE == SAI_NEXT_HOP_TYPE_SEGMENTROUTE_ENDPOINT
+     *** @deprecated true**
+     */
+    SAI_NEXT_HOP_ATTR_SEGMENTROUTE_ENDPOINT_POP_TYPE,
+
+...
+...
+
+} sai_next_hop_attr_t;
+```
+
+### sairouterinterface.h ###
+
+1. Added attribute to enable/disable SRv6 on a RIF.
```

Comment:
As provided earlier, unlike MPLS, per-intf enabling for SRv6 does not make sense. SRv6 should be enabled /diabled globally instead.